aspnetcore/security/anti-request-forgery.md

```diff
@@ -4,142 +4,230 @@ author: steve-smith
+
+### Token-based authentication
+
+Token-based authentication doesn't store session on the server. When a user is authenticated, they're issued a token (not an antiforgery token). The token contains user information in the form of [claims](/dotnet/framework/security/claims-based-identity-model). When a user attempts to access a resource requiring authentication, the token is sent to the app with an additional authorization header in form of Bearer token. This makes the app stateless. In each subsequent request, the token is passed in the request for server-side validation. This token isn't *encrypted*; it's *encoded*. On the server, the token is decoded to access its information. To send the token on subsequent requests, either store the token in the browser's local storage or in a cookie. Don't be concerned about CSRF vulnerability if the token is stored in the browser's local storage. CSRF is a concern when the token is stored in a cookie.
+
+### Multiple apps hosted at one domain
+
+Although `example1.cloudapp.net` and `example2.cloudapp.net` are different hosts, there's an implicit trust relationship between hosts under the `*.cloudapp.net` domain. This implicit trust relationship allows potentially untrusted hosts to affect each other's cookies (the same-origin policies that govern AJAX requests don't necessarily apply to HTTP cookies).
+
+The ASP.NET Core runtime provides some mitigation in that the username is embedded into the field token. Even if a malicious subdomain is able to overwrite a session token, it can't generate a valid field token for the user. When hosted in such an environment, the built-in anti-CSRF routines still can't defend against session hijacking or login CSRF attacks. Shared hosting environments are vulnerable to session hijacking, login CSRF, and other attacks.
+
+## ASP.NET Core antiforgery configuration 
+
+> [!WARNING]
+> ASP.NET Core implements antiforgery using [ASP.NET Core Data Protection](xref:security/data-protection/introduction). The data protection stack must be configured to work in a server farm. See [Configuring data protection](xref:security/data-protection/configuration/overview) for more information.
 
-In ASP.NET Core MVC 2.0 the [FormTagHelper](xref:mvc/views/working-with-forms#the-form-tag-helper) injects anti-forgery tokens for HTML form elements. For example, the following markup in a Razor file will automatically generate anti-forgery tokens:
+In ASP.NET Core MVC 2.0 or later, the [FormTagHelper](xref:mvc/views/working-with-forms#the-form-tag-helper) injects antiforgery tokens into HTML form elements. The following markup in a Razor file automatically generates antiforgery tokens:
```

Comment:
Make a note that the HtmlHelper always added the CSRF field. I believe this is the case @NTaylorMullen can confirm.

Reply:
HtmlHelper's BeginForm generates antiforgery by default if the forms `method` is not `get`